imagersite/imager_images/models.py

```diff
@@ -0,0 +1,57 @@
+from django.db import models
+from django.contrib.auth.models import User
+from django.utils.encoding import python_2_unicode_compatible
+
+PUBLISHED_CHOICES = (
+    ('private', 'private'),
+    ('shared', 'shared'),
+    ('public', 'public')
+)
+
+
+@python_2_unicode_compatible
+class Photos(models.Model):
+    image = models.ImageField(upload_to='photo_files/%Y-%m-%d')
+    user = models.ForeignKey(
+        User,
+        null=False
+    )
+    title = models.CharField(max_length=256)
+    description = models.TextField()
+    date_uploaded = models.DateField(auto_now_add=True)
+    date_modified = models.DateField(auto_now=True)
+    date_published = models.DateField(auto_now_add=True)
```

Comment:
I think this date should only be set when the `published` state of the file is changed to `public`.